api/debug.go

```diff
@@ -380,6 +266,163 @@ func (d *DebugAPI) FlowHeightByBlock(
 	return cdcHeight, nil
 }
 
+func (d *DebugAPI) traceTransaction(
```

Comment:
I have extracted the logic of `TraceTransaction` & `TraceBlockByNumber` to private functions, `traceTransaction` & `traceBlockByNumber` respectively, so that we can re-use their functionality in the `DebugAPI`, without actually applying any rate-limit. The rate-limit is applied to the public-facing functions.